src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Yarp.ReverseProxy.Forwarder;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    internal class DefaultHttpProxyService : IHttpProxyService
+    {
+        private IHttpForwarder _httpForwarder;
+        private HttpMessageInvoker _messageInvoker;
+        private ForwarderRequestConfig _forwarderRequestConfig;
+        private string _proxyEndpoint;
+
+        public DefaultHttpProxyService(IHttpForwarder httpForwarder)
+        {
+            _httpForwarder = httpForwarder;
+
+            var handler = new SocketsHttpHandler();
+            _messageInvoker = new HttpMessageInvoker(handler);
+            _forwarderRequestConfig = new ForwarderRequestConfig();
+
+            // TODO: Update this logic. Port should come through configuration.
+            var port = Environment.GetEnvironmentVariable("AZURE_FUNCTIONS_HTTP_PROXY_PORT") ?? "5555";
+
+            _proxyEndpoint = "http://localhost:" + port;
+        }
+
+        public ValueTask<ForwarderError> Forward(ScriptInvocationContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.Inputs is null)
+            {
+                throw new ArgumentNullException(nameof(context.Inputs));
```

Comment:
TBH, I feel ArgumentNullException here for the sub property (of the argument passed in) is not the best.  May be InvalidOperationException?